### PR TITLE
CF-611: Refactoring of bootstrap script

### PR DESCRIPTION
**Pull request by cschroer**
Add pre-commit hooks to
  * Format shellscripts
  * run shellcheck and double quote to prevent globbing / word splitting

Refactoring for keyless bootstrap

Remove terragrunt support

### CODE REVIEW

File: bootstrap/bootstrap.sh [hunk 1]
```diff
@@ -209,154 +189,241 @@ cat <<-EOF
...
 
-if [ "$MODE" = "terraform" ]
-then
-	echo "${TEXT_COLOR_MAGENTA}You need to import the created service account into your state:${TEXT_ALL_OFF}"
-	echo "terraform init && terraform import 'module.project-cfg.google_service_account.service_accounts[\"${SA_NAME}\"]' projects/${GCP_PROJECT_ID}/serviceAccounts/${SA_NAME}@${GCP_PROJECT_ID}.iam.gserviceaccount.com"
+	${TEXT_COLOR_MAGENTA}Generated IaC code files in '${OUTPUT_DIR}' - please review configuration!${TEXT_ALL_OFF}
 
-	echo "${TEXT_COLOR_MAGENTA}You need to import the created GCS bucket into your state:${TEXT_ALL_OFF}"
-	echo "terraform import 'module.tf-state-bucket.google_storage_bucket.bucket' ${GCP_PROJECT_ID}/${GCS_BUCKET}"
+	Once you reviewed the generated Infrastucre as Code configuration, the script supports to automaticly import
```
Comment (DatMicha) [inline bootstrap/bootstrap.sh:355]:
```suggestion
	Once you reviewed the generated Infrastructure as Code configuration, the script supports to automatically import
```

File: bootstrap/bootstrap.sh [hunk 2]
```diff
@@ -209,154 +189,241 @@ cat <<-EOF
...
+	the generaded resources into the terraform state. THis resouces include:
```
Comment (DatMicha) [inline bootstrap/bootstrap.sh:356]:
```suggestion
	the generadet resources into the terraform state. This includes the following resouces:
```

File: bootstrap/bootstrap.sh [hunk 3]
```diff
@@ -209,154 +189,241 @@ cat <<-EOF
...
+		* The IaC Service Account created to execute this code
+		* The Google Cloud Storage bucket created to store your remote state
+
+END_OF_DOC
+read -p "Proceed with import? (y/n)" -n 1 -r
+echo
+if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+  cat <<-END_OF_DOC
+
+		${TEXT_COLOR_MAGENTA}You may need to import the resouces manual your state:${TEXT_ALL_OFF}
```
Comment (DatMicha) [inline bootstrap/bootstrap.sh:366]:
```suggestion
		${TEXT_COLOR_MAGENTA}You may need to import the resources manually into your state:${TEXT_ALL_OFF}
```

File: bootstrap/bootstrap.sh [hunk 4]
```diff
@@ -209,154 +189,241 @@ cat <<-EOF
...
+  reset_command_output_trap
+
+  # Build plan
+  (rm -f bootstrap.tfplan && terraform plan -out bootstrap.tfplan >/dev/null 2>&1)
+  TF_PLAN=$(terraform show bootstrap.tfplan)
+  cat <<-END_OF_DOC
+
+		${TEXT_COLOR_MAGENTA}Execure terraform?${TEXT_ALL_OFF}
+
+		The script has done all the initial bootstrapping to execure terraform for the first time!
```
Comment (DatMicha) [inline bootstrap/bootstrap.sh:411]:
```suggestion
		The script has done all the initial bootstrapping to execute terraform for the first time!
```

File: bootstrap/bootstrap.sh [hunk 5]
```diff
@@ -209,154 +189,241 @@ cat <<-EOF
...
+		The module will create a lot of additional resources at the first run, depending on your configuration:
```
Comment (DatMicha) [inline bootstrap/bootstrap.sh:412]:
```suggestion
		The module will create several additional resources during the first run, depending on your configuration:
```

File: bootstrap/templates/project.tf.in [hunk 6]
```diff
@@ -14,52 +14,62 @@
...
+    ${GITHUB_REPOSITORY_IAM_BLOCK_STRING}
   }
 
   service_accounts = {
     "${SA_SHORT_NAME}" = {
       display_name = "Service account used in IaC pipelines"
-      iam          = {} # No special Service Account resource IAM permissions
+      iam          = {
+        "roles/iam.serviceAccountTokenCreator" = [
+          # Allow manager group ti impersonate IaC service account
```
Comment (DatMicha) [inline bootstrap/templates/project.tf.in:66]:
```suggestion
          # Allow manager group to impersonate IaC service account
```

**Review by DatMicha — CHANGES_REQUESTED** [review]
Haven't run the script, but found trivial improvements.

**Review by mariuskiessling — APPROVED** [review]
lgtm

Thank you for all the work!

**Pull request merged**